### DISCUSSION TITLE
Does the `Docs` tool belong in the dubugger?

### DISCUSSION THREAD

**Discussion by matt-y** [jpmonettas/flow-storm-debugger#64] [General]
[The docs tool that lives within the debugger GUI ](https://jpmonettas.github.io/flow-storm-debugger/user_guide.html#_docs_tool) feels out of place. I may not be understanding what this feature is really providing - but there are a myriad of tools for clojure that perform documentation generation. Forgive the pointed question but why do I need documentation generation functionality within my debugger? It seems wholly unrelated to debugging. 

What does the documentation tool here provide that - as a quick example -[ the `codox` tool ](https://github.com/weavejester/codox)does not provide?

Would the `Docs` tool make more sense as an "extension" to the flow storm debugger, rather than being part of the core tool? Or would it make sense to make the documentation tool wholly external to the debugger GUI?

**Comment by jpmonettas** [comment]
You are right that it doesn't make sense for something called a debugger to contain a documentation explorer, but here are my reasons : 

First this is what flow docs is about https://github.com/jpmonettas/flow-storm-debugger/blob/master/docs/flow_docs.md, I don't think there is any other tool currently providing that, for clojure or any other language. If you know about one I would love to know about it.

First reason is that the doc system was developed pretty close to the debugger, since it consists of instrumenting entire code bases, but not for stepping over it but instead for sampling it while running the tests and collecting information about types, call examples, etc.  

Second reason is that I don`t want to have multiple UIs open, one for debugging, another for checking documentation, so I temporarily added the functionality there while pushing the debugger more into a companion tool (a sidecar) for providing a bunch of tools to assist you with your repl flow.

I have also thought about creating an extension system, like a plugin system, but didn't had time to think a lot about it yet.

Does it make sense?

**Reply by matt-y** [reply]
Yes that makes sense; appreciate you even entertaining the question.
